### PR TITLE
Add support for expanded lead images

### PR DESCRIPTION
**Pull request by danielcopaciu**
In order to return expanded lead images, internal-content-api has to make a POST request to image-resolver which will return the entire content together with expanded images if no error occurs. If an error occurs, it should return the content without expanded lead images

### CODE REVIEW

File: handlers.go [hunk 1]
```diff
@@ -74,7 +95,10 @@ func (h internalContentHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
 			return
 		}
 	}
-	mergedContent := h.resolveAdditionalFields(parts, uuid)
+
+	mergedContent := mergeParts(parts)
```
Comment (radutudor) [inline handlers.go]: why was it needed to take out `mergeParts` and `resolveLeadImages`  from `resolveAdditionalFields` ?

File: handlers.go [hunk 2]
```diff
@@ -74,7 +95,10 @@ func (h internalContentHandler) ServeHTTP(w http.ResponseWriter, r *http.Request
...
+	mergedContent = h.resolveLeadImages(mergedContent, h.serviceConfig.envAPIHost, uuid, ctx, expandImages)
```
Comment (radutudor) [inline handlers.go]: go best practices state that the context should generally be the first param. Also, I feel we are not using the context fully. We are currently using it only to store de uuid. We could add more stuff to it (maybe the uuid? )

File: handlers.go [hunk 3]
```diff
@@ -163,6 +185,101 @@ func resolveLeadImageURLs(content map[string]interface{}, APIHost string) {
 		imgURL := "http://" + APIHost + "/content/" + img["id"].(string)
 		img["id"] = imgURL
 	}
+
+	transformedContent := make(map[string]interface{})
```
Comment (radutudor) [inline handlers.go]: I don't think this `transformedContent` + the copy loop below are needed.
If `expandImages` is false, return `content`, if `,true`, the `transformedContent` will be created when `getExpandedImages` is called.

File: handlers.go (hunk 1 above)
Comment (danielcopaciu) [inline handlers.go]: You are right, it is not. I was not paying attention to the fact that the handler is sent as a parameter to each function and I needed that. I fixed this by moving back the two functions and send the handler as a parameter to my function.

File: handlers.go (hunk 2 above)
Comment (danielcopaciu) [inline handlers.go]: I noticed that both the tid and the uuid are stored in context, so I changed my code to use this.

File: handlers.go (hunk 3 above)
Comment (danielcopaciu) [inline handlers.go]: Fixed

**Pull request merged**